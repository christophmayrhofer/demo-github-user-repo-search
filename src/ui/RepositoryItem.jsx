import React, { PropTypes } from 'react';

const RepositoryItem = ({ name, htmlUrl, description, language }) => (
  <div className="panel panel-default repositoryPanel">
    <div className="panel-heading">
      <a href={htmlUrl}> {name} </a>

      <If condition={language}>
        <b> ({language}) </b>
      </If>

    </div>
    <div className="panel-body"> <cite> {description} </cite> </div>
  </div>
);

RepositoryItem.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  htmlUrl: PropTypes.string.isRequired,
};

export default RepositoryItem;
