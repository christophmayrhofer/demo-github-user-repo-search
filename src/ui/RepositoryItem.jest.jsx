import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import RepositoryItem from './RepositoryItem';

describe('RepositoryItem', () => {
  it('renders correctly', () => {
    const props = {
      name: 'RepositoryName',
      language: 'RepositoryLanguage',
      description: 'RepositoryDescription',
      htmlUrl: 'RepositoryUrl',
    };

    const wrapper = shallow(
      <RepositoryItem {...props} />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
