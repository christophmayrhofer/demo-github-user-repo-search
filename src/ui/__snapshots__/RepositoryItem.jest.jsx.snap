exports[`RepositoryItem renders correctly 1`] = `
<div
  className="panel panel-default repositoryPanel">
  <div
    className="panel-heading">
    <a
      href="RepositoryUrl">
       
      RepositoryName
       
    </a>
    <b>
       (
      RepositoryLanguage
      ) 
    </b>
  </div>
  <div
    className="panel-body">
     
    <cite>
       
      RepositoryDescription
       
    </cite>
     
  </div>
</div>
`;
